### ISSUE TITLE
Turn airplane mode on then turn it off , turn wifi off and turn cellular on -> broken function connect()

### ISSUE THREAD

**Issue by hhsang** [biner88/mysql_utils#20]
1. wifi working fine both android and ios 
2. Cellular data fine on android but iOS 50% when I turn airplane mode on then off then turn cellular on (Wifi 100% off) I receive error: SocketException: Connection failed (OS Error: Network is unreachable, errno = 51), address = 11.111.222.33, port = 3306.
3. Flutter 3.16.9 and I build minimum iOS  >= 13.

**Comment by biner88** [comment]
This error is caused by the socket connection not being closed properly. It is a problem with Dart's underlying code. I recommend detecting a no-network state and actively closing the MySQL connection. Or use try catch?

**Comment by hhsang** [comment]
> This error is caused by the socket connection not being closed properly. It is a problem with Dart's underlying code. I recommend detecting a no-network state and actively closing the MySQL connection. Or use try catch?

Yes, I have detecting network, till working and I try call api json holder to test and return status 200 with correct response.

**Comment by hhsang** [comment]
If the app connect  success no issue when I using app , and if it had problem about connect  the app same dead (Although I restart devices) only working when I turn on/off airplane mode and turn on Cellular again.
